reference-implementation/to-upstream-wpts/writable-streams/write.js

```diff
@@ -84,42 +79,39 @@ async_test(t => {
 
   const writer = ws.getWriter();
 
-  setTimeout(t.step_func(() => {
-    assert_equals(writer.desiredSize, 1, 'desiredSize should be 1');
+  assert_equals(writer.desiredSize, 1, 'desiredSize should be 1');
 
-    writer.ready.then(t.step_func(() => {
-      const writePromise = writer.write('a');
-      let writePromiseResolved = false;
-      assert_not_equals(resolveSinkWritePromise, undefined, 'resolveSinkWritePromise should not be undefined');
+  writer.ready.then(() => {
```

Comment:
Yes. This is a bit of a gotcha with promise_test. But at least unlike the gotchas in async_test, it's consistent with how Promises work in general.